File: stone_lib/src/main/java/com/github/klee0kai/stone/annotations/component/GcAllScope.java
package com.github.klee0kai.stone.annotations.component;


import javax.inject.Scope;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * A scope that defines all exposed objects in DI.
 * Used for garbage collection and caching change methods.
 */

@GcScopeAnnotation
@Scope
@Retention(value = RetentionPolicy.CLASS)
@Target(value = ElementType.METHOD)
public @interface GcAllScope {
}

File: stone_multiplatform/src/commonMain/kotlin/stone/annotations/component/GcAllScope.kt
package stone.annotations.component

/**
 * A scope that defines all exposed objects in DI.
 * Used for garbage collection and caching change methods.
 */
@GcScopeAnnotation
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class GcAllScope 
